lib/vdsm/API.py

```diff
@@ -214,7 +215,15 @@ def create(self, vmParams):
                             self.log.info('loaded pickledMachineParams: %s',
                                           pickledMachineParams)
                             self.log.debug('former conf: %s', vmParams)
-                            vmParams.update(pickledMachineParams)
+                            modified_former_params = \
+                                cpumanagement.resume_cpu_pinning(
+                                    vmParams,
+                                    pickledMachineParams)
+                            if modified_former_params != pickledMachineParams:
```

Comment:
resume_cpu_pinning returns pickledMachineParams, which is modified by side-effect, so this condition is never satisfied.